pkg/server/server.go

```diff
@@ -433,6 +433,16 @@ func (s *ProxyServer) readFrontendToChannel(stream client.ProxyService_ProxyServ
 func (s *ProxyServer) serveRecvFrontend(stream client.ProxyService_ProxyServer, recvCh <-chan *client.Packet) {
 	klog.V(5).Infoln("start serving frontend stream")
 
+	defer func() {
+		// As the read side of the recvCh channel, we cannot close it.
+		// However readFrontendToChannel() may be blocked writing to the channel,
+		// so we need to consume the channel until it is closed.
+		for range recvCh {
+			// Ignore values as this indicates there was a problem
+			// with the remote connection.
+		}
```

Comment:
Can we add a debug message to let us know how many packets we dropped? Might even be worth adding a metric.

Reply:
Done. Also added to the agent version of this.